### PR TITLE
[#IOCIT-153] Add premium support to Third Party Messages

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes

- Add ThirdParty Support to `ApiNewMessageWithAdvancedFeatures`,
- Add unit tests
- Add Integration test
#### Motivation and Context

This is needed in order to provide Third Party Feature only to ADVANCED Messages.
#### How Has This Been Tested?

Unit and integration tests
#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: __integrations__/index.test.ts [hunk 1]
```diff
@@ -261,6 +261,35 @@ describe("Create Message |> Middleware errors", () => {
     });
   });
 
+  it("should return 403 when creating a no ADVANCED third party message with wrong permission", async () => {
+    const nodeFetch = getNodeFetch({
+      "x-user-groups":
+        customHeaders["x-user-groups"] + ",ApiMessageWriteAdvanced"
+    });
```
Comment (gquadrati) [inline __integrations__/index.test.ts:268]: A ThirdPartyMessage with `Standard` level returns 201.
Is this the intended behavior?

Comment (gquadrati) [reply __integrations__/index.test.ts:268]: Plus, tests says `when creating a no ADVANCED..`, but we added `ApiMessageWriteAdvanced` as header.

Comment (AleDore) [reply __integrations__/index.test.ts:268]: Nope, actually thirdPartyMessage with `Standard` level must return 403 if user doesn't have `ApiThirdPartyMessageWrite` group.

Comment (gquadrati) [reply __integrations__/index.test.ts:268]: Sure, With `ApiThirdPartyMessageWrite` of course (I missed this part, sorry).

**Review by gquadrati — APPROVED** [review]
LGTM

**Pull request merged**